app/src/main/java/com/example/nvdovin/weatherapp/MainActivity.java

```diff
@@ -1,80 +1,70 @@
 package com.example.nvdovin.weatherapp;
 
-import android.os.AsyncTask;
 import android.support.v7.app.AppCompatActivity;
 import android.os.Bundle;
-import android.telecom.Call;
 import android.widget.TextView;
+import android.widget.Toast;
 
 import com.example.nvdovin.weatherapp.factory.GreenDaoFactory;
 import com.example.nvdovin.weatherapp.factory.RetrofitFactory;
 import com.example.nvdovin.weatherapp.model.City;
-import com.example.nvdovin.weatherapp.model.FirstModel;
-
-import java.io.IOException;
-import java.util.ArrayList;
 import java.util.List;
 
-import retrofit2.Response;
-
-public class MainActivity extends AppCompatActivity {
-    private List<City> cities;
+public class MainActivity extends AppCompatActivity implements MainView {
     private String query_city_name = "Chisinau";
     RetrofitFactory retrofitFactory;
     GreenDaoFactory greenDaoFactory;
 
+    MainPresenter mainPresenter;

```

Comment:
what does "MainPresenter" mean? Is it really "Main"?Do you plan inherit his logic? Or does it keep main logic for something? Please use meaningful class name. the same for MainView.